### PR TITLE
Revitalize all

### PR DESCRIPTION
**Pull request by if0s**
https://github.com/elasticio/microsoft-onedrive-component/issues/46
https://github.com/elasticio/microsoft-onedrive-component/issues/47
https://github.com/elasticio/microsoft-onedrive-component/issues/48
https://github.com/elasticio/microsoft-onedrive-component/issues/49
https://github.com/elasticio/microsoft-onedrive-component/issues/50

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,42 +1,51 @@
...
+* New authentication mechanism
+* Added retries on server errors
+* Added retries on 429 (too many requests) errors
+* Attachment URL provided directly in messages
+* Added support for uploading large files to OneDrive
+* Updated ReadMe 
+
+## 1.0.6 (September 23, 2022)
+
+* Update Sailor version to 2.6.29
```
Comment (A3a3e1) [inline CHANGELOG.md:12]:
```suggestion
* Update Sailor version to 2.7.1

```

File: component.json [hunk 2]
```diff
@@ -1,169 +1,122 @@
...
-      "title": "Get File",
-      "note": "Action to get file by provided path",
-      "main": "./lib/actions/getFile.js",
+    },
+    "deleteFile": {
+      "main": "./src/actions/deleteFile.js",
+      "title": "Delete File",
+      "help": {
+        "description": "",
+        "link": "/components/microsoft-onedrive/index.html#delete-file"
```
Comment (A3a3e1) [inline component.json:56]:
```suggestion
        "link": "/components/onedrive/index.html#delete-file"
```

File: component.json [hunk 3]
```diff
@@ -1,169 +1,122 @@
...
-    "deleteItem": {
-      "title": "Delete File",
-      "note": "Action to delete file by provided path",
-      "main": "./lib/actions/deleteItem.js",
+    "lookupObject": {
+      "main": "./src/actions/lookupObject.js",
+      "title": "Get File",
+      "help": {
+        "description": "Get File",
+        "link": "/components/microsoft-onedrive/index.html#get-file"
```
Comment (A3a3e1) [inline component.json:93]:
```suggestion
        "link": "/components/onedrive/index.html#get-file"
```

File: component.json [hunk 4]
```diff
@@ -1,169 +1,122 @@
...
     },
-    "upsertFile": {
+    "upsertObject": {
+      "main": "./src/actions/upsertObject.js",
       "title": "Upsert File",
-      "note": "Action upserts (create or replace) file by path",
-      "main": "./lib/actions/upsertFile.js",
+      "help": {
+        "description": "Upsert File",
+        "link": "/components/microsoft-onedrive/index.html#upsert-file"
```
Comment (A3a3e1) [inline component.json:118]:
```suggestion
        "link": "/components/onedrive/index.html#upsert-file"
```

File: component.json [hunk 5]
```diff
@@ -174,58 +127,116 @@
...
+    }
+  },
+  "triggers": {
+    "getNewAndUpdatedObjectsPolling": {
+      "main": "./src/triggers/getNewAndUpdatedObjectsPolling.js",
+      "title": "Get New and Updated Objects Polling",
+      "type": "polling",
+      "help": {
+        "description": "Poll for new and/or updated objects",
+        "link": "/components/microsoft-onedrive/index.html#get-new-and-updated-objects-polling"
```
Comment (A3a3e1) [inline component.json:142]:
```suggestion
        "link": "/components/onedrive/index.html#get-new-and-updated-objects-polling"
```

**Review by A3a3e1 — APPROVED** [review]

**Comment by HannaTrotsenko** [comment]
- [ ] add action's description to the hits
[Запис з 19.05.23 09:58:42.webm](https://github.com/elasticio/microsoft-onedrive-component/assets/28867440/ffcbe7d3-bd57-4e88-8063-66e5016b356c)
- [ ] qa

**Comment by HannaTrotsenko** [comment]
- [ ] Delete file action: handle case for retrieve sample in case `emit nothing` is choosen and file doesn't exist, bacause infinity sample
image

- [ ] qa

**Review by HannaTrotsenko — APPROVED** [review]

**Pull request merged**